src/jsonrpc/client/utils.rs

```diff
@@ -0,0 +1,197 @@
+        if rc != 0 {
+            Err($crate::stratis::StratisError::Error(rs))
+        } else if !changed {
+            Err($crate::stratis::StratisError::Error(
+                "The requested action had no effect".to_string(),
+            ))
+        } else {
+            Ok(())
+        }
+    }};
+}
+
+#[macro_export]
+macro_rules! left_align {
+    ($string:expr, $max_length:expr) => {{
+        let len = $string.len();
+        $string + vec![" "; $max_length - len + 3].join("").as_str()
+    }};
+}
+
+#[macro_export]
+macro_rules! right_align {
+    ($string:expr, $max_length:expr) => {
+        vec![" "; $max_length - $string.len() + 3].join("") + $string.as_str()
+    };
+}
+
+#[macro_export]
+macro_rules! align {
+    ($string:expr, $max_length:expr, $align:tt) => {
+        if $align == ">" {
+            $crate::right_align!($string, $max_length)
+        } else {
+            $crate::left_align!($string, $max_length)
+        }
+    };
+}
+
+#[macro_export]
+macro_rules! print_table {
+    ($($heading:expr, $values:expr, $align:tt);*) => {{
+        let (lengths_same, lengths) = vec![$($values.len()),*]
+            .into_iter()
+            .fold((true, None), |(is_same, len_opt), len| {
+                if len_opt.is_none() {
+                    (true, Some(len))
+                } else {
+                    (is_same && len_opt == Some(len), len_opt)
+                }
+            });
+        if !lengths_same {
+            return Err($crate::stratis::StratisError::Error(
+                "All values parameters must be the same length".to_string()
+            ));
+        }
+        let mut output = vec![String::new(); lengths.unwrap_or(0) + 1];
+        $(
+            let max_length = $values
+                .iter()
+                .fold($heading.len(), |acc, val| {
+                    if val.len() > acc {
+                        val.len()
+                    } else {
+                        acc
+                    }
+                });
+            if let Some(string) = output.get_mut(0) {
+                string.push_str($crate::align!($heading.to_string(), max_length, $align).as_str());
+            }
+            for (index, row_seg) in $values.into_iter()
+                .map(|s| $crate::align!(s, max_length, $align))
+                .enumerate()
+            {
+                if let Some(string) = output.get_mut(index + 1) {
+                    string.push_str(row_seg.as_str());
+                }
+            }
+        )*
+        for row in output.into_iter() {
+            println!("{}", row);
+        }
+    }};
+}
+
+const SUFFIXES: &[(u64, &str)] = &[
+    (60, "EiB"),
+    (50, "PiB"),
+    (40, "TiB"),
+    (30, "GiB"),
+    (20, "MiB"),
+    (10, "KiB"),
+    (1, "B"),
+];
+
+#[allow(clippy::cast_precision_loss)]
+pub fn to_suffix_repr(size: u128) -> String {
+    SUFFIXES.iter().fold(String::new(), |acc, (div, suffix)| {
+        let div_shifted = 1 << div;
```

Comment:
```acc``` ends up as empty string if size is 0.

Reply:
There may also be a problem with the (1, "B") value too. I think it should be (0, "B"). I'll clean this up.